core/src/main/scala/chisel3/internal/firrtl/Serializer.scala

```diff
@@ -297,6 +297,17 @@ private[chisel3] object Serializer {
       serializeIntrinsic(ctx, info, None, None, intrinsic, args, params, typeAliases)
     case i @ DefIntrinsicExpr(info, intrinsic, id, args, params) =>
       serializeIntrinsic(ctx, info, Some(i.name), Some(id), intrinsic, args, params, typeAliases)
+    case FirrtlComment(text) =>
+      // Because of split, iterator will always be non-empty even if text is empty
+      val it = text.split("\n").iterator
+      while ({
+        val line = it.next()
+        b ++= "; "; b ++= line
+        if (it.hasNext) {
+          newLineAndIndent()
+        }
+        it.hasNext
+      }) ()
```

Comment:
I know this is the correct approach given that Scala dropped `do--while`. It still looks weird to me, though... 🫠

Reply:
I don't love it but it is the most direct way to write this and I do prefer it to duplicating logic for first line vs. the rest 🤷‍♂️